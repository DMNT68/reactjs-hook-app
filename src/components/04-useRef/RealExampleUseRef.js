import React, { useState } from 'react'
import { MultipleCustomHokks } from '../03-examples/MultipleCustomHokks'

import '../02-useEffect/effects.css'

export const RealExampleUseRef = () => {
   
    const [show, setShow] = useState(false); 

    return (
        <div>
            <h1>RealExampleUseRef</h1>
            <hr/>

            {show && <MultipleCustomHokks />}
        
            <button className="btn btn-outline-primary mt-5" onClick={()=> setShow(!show)} >Show/Hidde</button>

        </div>
    )
}
