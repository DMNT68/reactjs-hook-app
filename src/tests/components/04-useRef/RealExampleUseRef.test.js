import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleUseRef } from '../../../components/04-useRef/RealExampleUseRef';
import { MultipleCustomHokks } from '../../../components/03-examples/MultipleCustomHokks';
describe('Pruebas en <RealExampleUseRef />', () => {

    const wrapper = shallow( <RealExampleUseRef /> );
    test('debe de mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHokks').exists()).toBe(false);
        
    });

    test('debe de mostrarse el componente <MultipleCustomHokks />', () => {
        
        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultipleCustomHokks').exists() ).toBe(true);
        
    });

});
