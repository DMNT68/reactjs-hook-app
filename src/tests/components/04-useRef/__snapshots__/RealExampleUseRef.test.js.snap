// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleUseRef /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    RealExampleUseRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-5"
    onClick={[Function]}
  >
    Show/Hidde
  </button>
</div>
`;
